### PR TITLE
too few braces for f strings to work

### LINKED ISSUE #51: Issue with downloading FITS files
When FITS files are downloaded, the filename is set with the string "HARPNUM" instead of the actual HARP number, as was done in previous versions. The version I am using is 0.6.0.

Current output (version 0.6.0):
-------------------------------------------------------------------------------------------------
Downloading file 1 of 10...
    record: hmi.sharp_cea_720s[5127][2015.02.04_01:00:00_TAI]
  filename: hmi.sharp_cea_720s.HARPNUM.20150204_010000_TAI.magnetogram.fits
-------------------------------------------------------------------------------------------------

Old output (previous versions):
-------------------------------------------------------------------------------------------------
Downloading file 1 of 10...
    record: hmi.sharp_cea_720s[5127][2015.02.04_01:00:00_TAI]
  filename: hmi.sharp_cea_720s.5127.20150204_010000_TAI.magnetogram.fits
-------------------------------------------------------------------------------------------------

### PR DESCRIPTION
**Pull request by nabobalis**
### Description

Fixes #51

### CODE REVIEW

File: drms/client.py [hunk 1]
```diff
@@ -682,7 +682,7 @@ def _generate_filenamefmt(self, sname):
             if si.keywords.loc[k].is_time:
                 pkfmt_list.append(f'{{{k}:A}}')
             else:
-                pkfmt_list.append(f'{k}')
+                pkfmt_list.append(f'{{{k}}}')
```
Comment (nabobalis) [inline drms/client.py:685]: My fault when I ported to the f strings. 

Checking the examples will be the way to check if this worked.

File: drms/utils.py [hunk 2]
```diff
@@ -68,7 +68,7 @@ def to_datetime(tstr, force=False):
     if force or s.str.endswith('_TAI').any():
         s = s.str.replace('_TAI', "")
         s = s.str.replace('_', ' ')
-        s = s.str.replace('.', '-', n=2)
+        s = s.str.replace('.', '-', regex=True, n=2)
```
Comment (nabobalis) [inline drms/utils.py:71]: This silences a future warning.

**Comment by nabobalis** [comment]
Looking at this example:
https://sunpy--52.org.readthedocs.build/projects/drms/en/52/generated/gallery/export_jpg.html#sphx-glr-generated-gallery-export-jpg-py

The 304 is not returned in the jpg file name and I am unsure if it should be?

File: tox.ini [hunk 3]
```diff
@@ -50,7 +50,7 @@ extras =
 commands =
     sunpy: pytest -vvv -s -ra --pyargs sunpy.net.jsoc --remote-data=any {posargs}
     !online: {env:PYTEST_COMMAND} {posargs}
-    online: {env:PYTEST_COMMAND} --reruns 2 --timeout=180 --remote-data=any --email nabil.freij@gmail.com {posargs}
+    online: {env:PYTEST_COMMAND} --reruns 2 --reruns-delay 30 --timeout=180 --remote-data=any --email nabil.freij@gmail.com {posargs}
```
Comment (Cadair) [inline tox.ini:53]: Dont you want to use the test email here?

**Review by Cadair — APPROVED** [review]

File: tox.ini (hunk 3 above)
Comment (nabobalis) [reply tox.ini]: Woops.

File: drms/json.py [hunk 4]
```diff
@@ -368,24 +368,24 @@ def _exp_request_url(
             method = 'url'
 
         if protocol in img_protocol_list:
-            d = {'ct': 'grey.sao', 'scaling': 'MINMAX', 'size': 1}
+            extra_keys = {'ct': 'grey.sao', 'scaling': 'MINMAX', 'size': 1}
```
Comment (nabobalis) [inline drms/json.py:371]: I was confused if this was extra d below or something else, so I renamed it.

File: drms/json.py [hunk 5]
```diff
@@ -368,24 +368,24 @@ def _exp_request_url(
...
-                protocol += f',min={d["min"]:g}'
-            if 'max' in d:
-                protocol += f',max={d["max"]:g}'
+            protocol += ',CT={ct},scaling={scaling},size={size}'.format(**extra_keys)
+            if 'min' in extra_keys:
+                protocol += f',min={extra_keys["min"]:g}'
+            if 'max' in extra_keys:
+                protocol += f',max={extra_keys["max"]:g}'
         else:
             if protocol_args is not None:
```
Comment (nabobalis) [inline drms/json.py:390]: I wonder if we make this raise a warning instead?

**Comment by nabobalis** [comment]
Sunpy failure is known doctest issue.

File: drms/json.py (hunk 4 above)
Comment (wtbarnes) [reply drms/json.py:371]: I'm confused by this comment.

**Review by wtbarnes — APPROVED** [review]

**Comment by wtbarnes** [comment]
> Sunpy failure is known doctest issue.

I kicked the tests and now I'm getting a different set of failures. Is this still true?

**Comment by nabobalis** [comment]
> > Sunpy failure is known doctest issue.
> 
> I kicked the tests and now I'm getting a different set of failures. Is this still true?

Should be.

File: drms/json.py (hunk 4 above)
Comment (nabobalis) [reply drms/json.py:371]: The var name being `d` confused with me with the code below that also uses a var called `d`.

**Comment by nabobalis** [comment]
> Looking at this example:
> [sunpy--52.org.readthedocs.build/projects/drms/en/52/generated/gallery/export_jpg.html#sphx-glr-generated-gallery-export-jpg-py](https://sunpy--52.org.readthedocs.build/projects/drms/en/52/generated/gallery/export_jpg.html#sphx-glr-generated-gallery-export-jpg-py)
> 
> The 304 is not returned in the jpg file name and I am unsure if it should be?

Open question

**Comment by wtbarnes** [comment]
> > Looking at this example:
> 
> > [sunpy--52.org.readthedocs.build/projects/drms/en/52/generated/gallery/export_jpg.html#sphx-glr-generated-gallery-export-jpg-py](https://sunpy--52.org.readthedocs.build/projects/drms/en/52/generated/gallery/export_jpg.html#sphx-glr-generated-gallery-export-jpg-py)
> 
> > 
> 
> > The 304 is not returned in the jpg file name and I am unsure if it should be?
> 
> 
> 
> Open question

The whole file name format for the jpg export looks different to me than the format for a FITS export

**Comment by nabobalis** [comment]
So we can ignore it?

**Pull request merged**